### DISCUSSION TITLE
Production Build Error Could Not Resolve "Crypto"

### DISCUSSION THREAD

**Discussion by Pirashan** [vercel/community#997] [Help] [answered]
Project works correctly on localhost, but the following errors are being thrown when I try to deploy.

Build failed with 3 errors:
--
16:52:08.252 | vc-file-system:node_modules/uuid/dist/esm-node/md5.js:1:19: ERROR: Could not resolve "crypto"
16:52:08.252 | vc-file-system:node_modules/uuid/dist/esm-node/rng.js:1:19: ERROR: Could not resolve "crypto"
16:52:08.252 | vc-file-system:node_modules/uuid/dist/esm-node/sha1.js:1:19: ERROR: Could not resolve "crypto"

My package.json looks like this:
{
  "name": "spotify2.0",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@heroicons/react": "^2.0.12",
    "crypto-browserify": "^3.12.0",
    "crypto-es": "^1.2.7",
    "crypto-js": "^4.1.1",
    "lodash": "^4.17.21",
    "next": "12.3.1",
    "next-auth": "^4.14.0",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "recoil": "^0.7.6",
    "spotify-web-api-node": "^5.0.2",
    "stream": "^0.0.2",
    "tailwind-scrollbar-hide": "^1.1.7"
  },
  "devDependencies": {
    "autoprefixer": "^10.4.12",
    "crypto": "^1.0.1",
    "eslint": "8.25.0",
    "eslint-config-next": "12.3.1",
    "postcss": "^8.4.18",
    "tailwindcss": "^3.1.8",
    "webpack": "^5.74.0"
  },
  "browser": {
    "crypto": false
  }
}

Not sure how to move forward, I looked into browserify, possibly using webpack, I added the "crypto": false portion and have no clue where to go from here. There are similar discussions but all the solutions I have found don't seem to work for me.

**Comment by amyegan** [comment]
Thanks for reaching out @Pirashan! If you can share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), I'm happy to help with debugging. Otherwise, I can suggest a few things to try:
- [change dependency names](https://stackoverflow.com/a/73258330)
- [add React modules that might be missing](https://github.com/web3/web3.js#web3-and-create-react-app)
- [modify tsconfig.json](https://stackoverflow.com/a/63829543) or [webpack config](https://stackoverflow.com/a/68400891)
- [try a different crypto-js version](https://stackoverflow.com/a/65506584)

Sorry if you have already tried these! I'm not sure what you have already done.

**Comment by agustinjch** [comment, marked as answer]
I found the issue, at least on my side. I didn't set up that the deployment was a next.js app, and it was in "Build and development settings" set as "Other", so when the deploy was done, it was giving me that error.

**Reply by Pirashan** [reply]
Worked for me, thank you so much!!
